modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageServiceTests.java

```diff
@@ -118,78 +115,38 @@ public void testReinitClientSettings() throws Exception {
+            final var client21 = storageService.client("gcs1", "repo1", statsCollector);
             assertThat(client21.getOptions().getProjectId(), equalTo("project_gcs21"));
             // old client 2 not changed
             assertThat(client12.getOptions().getProjectId(), equalTo("project_gcs12"));
             // new client2 is gone
             final IllegalArgumentException e2 = expectThrows(
                 IllegalArgumentException.class,
-                () -> storageService.client("gcs2", "repo2", randomPurpose(), statsCollector)
+                () -> storageService.client("gcs2", "repo2", statsCollector)
             );
             assertThat(e2.getMessage(), containsString("Unknown client name [gcs2]."));
             // client 3 emerged
-            final Storage client23 = storageService.client("gcs3", "repo3", randomPurpose(), statsCollector);
+            final var client23 = storageService.client("gcs3", "repo3", statsCollector);
             assertThat(client23.getOptions().getProjectId(), equalTo("project_gcs23"));
         }
     }
 
-    public void testClientsAreNotSharedAcrossRepositories() throws Exception {
-        final MockSecureSettings secureSettings1 = new MockSecureSettings();
-        secureSettings1.setFile("gcs.client.gcs1.credentials_file", serviceAccountFileContent("test_project"));
-        final Settings settings = Settings.builder().setSecureSettings(secureSettings1).build();
-        try (GoogleCloudStoragePlugin plugin = new GoogleCloudStoragePlugin(settings)) {
-            final GoogleCloudStorageService storageService = plugin.storageService;
-
-            final OperationPurpose operationPurpose = randomPurpose();
-            final OperationPurpose differentOperationPurpose = randomValueOtherThan(operationPurpose, BlobStoreTestUtil::randomPurpose);
-            final Storage repo1Client = storageService.client(
-                "gcs1",
-                "repo1",
-                operationPurpose,
-                new GoogleCloudStorageOperationsStats("bucket")
-            );
-            final Storage repo1ClientOtherPurpose = storageService.client(
-                "gcs1",
-                "repo1",
-                differentOperationPurpose,
-                new GoogleCloudStorageOperationsStats("bucket")
-            );
-            final Storage repo2Client = storageService.client(
-                "gcs1",
-                "repo2",
-                operationPurpose,
-                new GoogleCloudStorageOperationsStats("bucket")
-            );
-            final Storage repo1ClientSecondInstance = storageService.client(
-                "gcs1",
-                "repo1",
-                operationPurpose,
-                new GoogleCloudStorageOperationsStats("bucket")
-            );
-
-            assertNotSame(repo1Client, repo2Client);
-            assertNotSame(repo1Client, repo1ClientOtherPurpose);
-            assertSame(repo1Client, repo1ClientSecondInstance);
-        }
-    }
```

Comment:
I think this is still a valid test, right? we still don't want to share clients across repositories. We can just take all the operationpurpose stuff out of it.